src/read/macho/dyld_cache.rs

```diff
@@ -0,0 +1,214 @@
+        } else {
+            Endianness::Big
+        }
+    }
+
+    /// Return true if the file is little endian, false if it is big endian.
+    pub fn is_little_endian(&self) -> bool {
+        self.endian.is_little_endian()
+    }
+
+    /// Return true if the file can contain 64-bit addresses.
+    pub fn is_64(&self) -> bool {
+        self.is_64
+    }
+
+    /// Iterate over the images in this cache.
+    pub fn iter_images<'cache>(&'cache self) -> DyldCacheImageIterator<'data, 'cache, E, R> {
+        let images_offset = self.header.images_offset.get(self.endian) as u64;
+        let images_count = self.header.images_count.get(self.endian);
+        DyldCacheImageIterator {
+            cache: self,
+            images_count,
+            next_image_index: 0,
+            next_image_offset: images_offset,
+        }
+    }
+}
+
+/// An iterator over all the images (dylibs) in the dyld shared cache.
+#[derive(Debug)]
+pub struct DyldCacheImageIterator<'data, 'cache, E = Endianness, R = &'data [u8]>
+where
+    E: Endian,
+    R: ReadRef<'data>,
+{
+    cache: &'cache DyldCache<'data, E, R>,
+    images_count: u32,
+    next_image_index: u32,
+    next_image_offset: u64,
+}
+
+impl<'data, 'cache, E, R> DyldCacheImageIterator<'data, 'cache, E, R>
+where
+    E: Endian,
+    R: ReadRef<'data>,
+{
+    /// Advance the iterator and return the current image.
+    pub fn next(&mut self) -> Result<Option<DyldCacheImage<'data, E, R>>> {
+        if self.next_image_index >= self.images_count {
+            return Ok(None);
+        }
+        self.next_image_index += 1;
+        let data = self.cache.data;
+        let image_info = data
+            .read::<macho::DyldCacheImageInfo<E>>(&mut self.next_image_offset)
+            .read_error("Couldn't read macho::DyldCacheImageInfo")?;
+        Ok(Some(DyldCacheImage {
+            endian: self.cache.endian,
+            is_64: self.cache.is_64,
+            data,
+            first_mapping_address: self.cache.first_mapping_address,
+            image_info,
+        }))
+    }
+}
+
+/// One image (dylib) from inside the dyld shared cache.
+#[derive(Debug)]
+pub struct DyldCacheImage<'data, E = Endianness, R = &'data [u8]>
+where
+    E: Endian,
+    R: ReadRef<'data>,
+{
+    endian: E,
+    is_64: bool,
+    data: R,
+    first_mapping_address: u64,
+    image_info: &'data macho::DyldCacheImageInfo<E>,
+}
+
+impl<'data, E, R> DyldCacheImage<'data, E, R>
+where
+    E: Endian,
+    R: ReadRef<'data>,
+{
+    /// The file system path of this image.
+    pub fn path(&self) -> Result<&'data str> {
+        // The longest path I've seen is 164 bytes long. In theory paths could be longer than 256.
+        const MAX_PATH_LEN: u64 = 256;
+
+        let path_offset = self.image_info.path_file_offset.get(self.endian) as u64;
+        let slice_containing_path = self
+            .data
+            .read_bytes_at(path_offset, MAX_PATH_LEN)
+            .read_error("Couldn't read path")?;
```

Comment:
What do you think about adding a `ReadRef::read_string_at` or something like that?

Reply:
Sounds good to me.